fix(electron-updater): disable differential download

axel7083 opened the pull request:
### What does this PR do?

`electron-update` package uses differential download for the update. This is a problem because it consume a lot of memory to make a difference between the previous blockmap and the new one.

**We are pretty short in memory** today with Podman Desktop. This is not a _nice_ solution, as this may induce slight bigger download time (we download the full file instead of partial) but **this fixes the memory issue**.

Being able to download the update without crash in a longer time is better than nothing.

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->

### What issues does this PR fix or reference?

Should fixes the following issues
- https://github.com/podman-desktop/podman-desktop/issues/12035
- https://github.com/podman-desktop/podman-desktop/issues/10725

### How to test this PR?

<!-- Please explain steps to verify the functionality,
do not forget to provide unit/component tests -->

- [x] Tests are covering the bug fix or the new feature

1. checkout https://github.com/podman-desktop/podman-desktop/tree/1.17.x
2. merge this PR changes in the 1.17.x branch
3. `pnpm install`
4. `pnpm run compile:current`
5. execute `dist/*-unpacked/Podman Desktop.exe`
6. assert podman desktop starts
7. assert an update is available
8. start the update workflow

odockal reviewed (approved):
Built from main, changed version in package json to 1.17.0, applied the changes, build+compiled, tested, works on Windows 10 while without patch it crashes. I also think that there is missing very annoying glitch that blocked all UI while the blockmap was computed.

benoitf commented:
I would enable it only on Windows ( I don't recall errors on macOS)
And a hidden flag to change the behaviour in case it has unexpected drawbacks

amisskii reviewed (approved):
Tested on mac. It works fine. LGTM

axel7083 commented:
> I would enable it only on Windows ( I don't recall errors on macOS) And a hidden flag to change the behaviour in case it has unexpected drawbacks

@benoitf I updated the code and added a new configuration property `preferences.update.disableDifferentialDownload`:

- defaulting to `true` on window
- false on other platforms

axel7083 commented:
@Mergifyio backport 1.18.x
